design-proposals/sdn-ipam-secondary-nets.md

```diff
@@ -0,0 +1,450 @@
+
+Finally, OVN-Kubernetes must react to the deletion of the IPAMClaim, and return
+this IP address to its in-memory pool.
+
+### KubeVirt related changes
+
+#### Pod templating
+
+KubeVirt would need to template the pod accordingly - i.e. the
+`NetworkSelectionElements` must feature the `ipam-claim-reference` attribute to
+indicate to the IPAM CNI / CNI which `IPAMClaim` they should use to persist the
+IP allocation across reboots / migrations.
+
+Check the following network selection elements annotation from a KubeVirt
+launcher pod to see an example:
+
+```bash
+kubectl get pods <launcher pod name> -ojsonpath="{ @.metadata.annotations.k8s\.v1\.cni\.cncf\.io\/networks }" | jq
+[
+  {
+    "name": "tenantred",
+    "namespace": "default",
+    "interface": "pod16367aacb67",
+    "ipam-claim-reference": "vm-a.tenantred"
+  }
+]
+```
+
+#### Creating the IPAMClaim
+KubeVirt must create the IPAMClaims (and set finalizers / owner references) to
+properly ensure the VM's lifecycle is reflected in the allocation.
+
+There are two important sections to fill when creating the IPAMClaim reference:
+- the IPAMClaim name
+- the spec.network attribute
+
+The `IPAMClaim` name will be created using the following pattern:
+`<vm name>.<logical network name>`. The logical network name is the [name of the
+network](https://kubevirt.io/api-reference/main/definitions.html#_v1_network)
+in the KubeVirt API.
+
+The `IPAMClaim.spec.network` must be the name of network as per the
+[CNI spec](https://github.com/containernetworking/cni/blob/main/SPEC.md#configuration-format).
+This attribute is mandatory as per the CNI specification, and the CNI requires
+it to understand how to map IPAMClaims to the networks.
+
+Below you can find an example of a network attachment definition, specifying
+the behavior of attachments to the `tenantblue` network, along with an
+`IPAMCLaim` for that network.
+```yaml
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenantblue
+spec:
+  config: |2
+    {
+            "cniVersion": "0.4.0",
+            "name": "tenantblue",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"localnet",
+            "subnets": "192.168.10.0/24",
+            "excludeSubnets": "192.168.10.1/32",
+            "allowPersistentIPs": true,
+            "netAttachDefName": "default/tenantblue"
+    }
+---
+apiVersion: k8s.cni.cncf.io/v1alpha1
+kind: IPAMClaim
+metadata:
+  name: vm1.tenantblue
+spec:
+  network: tenantblue
+  interface: net1
+```
+
+**NOTE:** the mandatory `name` attribute in the CNI configuration
+(NAD.spec.config) **must** match the `IPAMClaim.spec.network` attribute of the
+IPAMClaim so the CNI plugin can map the IPAMClaim to the network.
+
+#### KubeVirt feature gate
+The feature is opt-in via the network-attachment-definition, (thus disabled by
+default) and it does not impose any performance costs when disabled; as a
+result, adding a feature gate is not recommended.
```

Comment:
This seems to confuse between feature lifecycle and the operational control of a feature.
A FG is required due to other reasons but if the configuration parameter is needed, it needs to be specified here with its default setting.

Reply:
Isn't "The feature is opt-in via the network-attachment-definition, (thus disabled bydefault)" basically that ? 

Furthermore, I've already thoroughly indicated how it works in the `Configuring the feature` section. Doing it again here feels like repeating myself.

Reply:
This is a FG section and you describe an operational configuration parameter.
I tried to explain the difference so you could decide how to adjust it.

Reply:
Whoever creates the IPAMClaims **must** indicate for which network does the claim correspond. Without that, the CNI's IP pool cannot be reconciled.

Reply:
I agree, I was talking about the FG. If it is disabled there is no need to parse the NAD config.

Reply:
Ah !!! My bad. 

Right.